syweb/webclient/app/settings/settings-controller.js

```diff
@@ -304,17 +392,29 @@ function($scope, matrixService, modelService, eventHandlerService, notificationS
             notificationService.clearRulesCache();
             fetchRules();
         });
+        rule.inprogress = true;
+    };
+
+    $scope.updateRuleEnabled = function(rule) {
+        rule.inprogress = true;
+        matrixService.setPushRuleEnabled('global', rule.kind, rule.rule_id, rule.enabled).then(function() {
+            notificationService.clearRulesCache();
+            fetchRules();
+        });
```

Comment:
It would be worth adding an error handler here too, even if it is just to log it.